### PR TITLE
Adding libnetwork support to calico

### PR DESCRIPTION
**Pull request by wavesoft**
## High-level description

This PR continues the work from https://github.com/mainred/dcos/pull/1 and adds `libnetwork` support to DC/OS calico networking. 

It also extends the e2e tests to cover the case of docker network in calico.

## Corresponding DC/OS tickets (required)

  - [D2IQ-58584](https://jira.d2iq.com/browse/D2IQ-58584) Add libnetwork (docker) support to calico 
  - [D2IQ-62588](https://jira.d2iq.com/browse/D2IQ-62588) Add e2e tests for calico 

## Related tickets (optional)

_None_

### CODE REVIEW

**Comment by wavesoft** [comment]
@mesosphere-mergebot request-review

**Comment by wavesoft** [comment]
@mesosphere-mergebot label Ready for review

File: packages/calico/extra/create-calico-docker-network.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env python
+"""
+This script creates the calico docker network if not exist, and configures
```
Comment (jkoelker) [inline packages/calico/extra/create-calico-docker-network.py]:
```suggestion
Creates the calico docker network if it does not exist, and configures
```

Comment (jkoelker) [reply packages/calico/extra/create-calico-docker-network.py]: Couldn't do a multi line suggestion, so reworded the beginning to keep the line length under 80 ;).

File: packages/calico/buildinfo.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "sources": {
     "calico": {
       "kind": "url",
-      "url": "https://github.com/projectcalico/cni-plugin/releases/download/v3.8.2/calico-amd64", 
+      "url": "https://github.com/projectcalico/cni-plugin/releases/download/v3.8.2/calico-amd64",
```
Comment (mainred) [inline packages/calico/buildinfo.json]: ❤️

File: packages/dcos-integration-test/extra/test_networking.py [hunk 3]
```diff
@@ -472,7 +472,7 @@ def vip_workload_test(dcos_api_session, container, vip_net, proxy_net, ipv6,
     # the network mode.
     # it is used for user network mode only
     if ipv6:
-        network_name = 'dcos6'
+        network_name = '{}6'.format(network_name)
```
Comment (mainred) [inline packages/dcos-integration-test/extra/test_networking.py]: We may make a note here that calico ipv6 is not supported yet.

File: packages/dcos-integration-test/extra/test_composition.py [hunk 4]
```diff
@@ -168,9 +168,11 @@ def test_systemd_units_are_healthy(dcos_api_session) -> None:
         'dcos-telegraf.socket',
         'dcos-fluent-bit.service']
     slave_units = [
+        'dcos-calico-libnetwork-plugin.service',
```
Comment (mainred) [inline packages/dcos-integration-test/extra/test_composition.py]: several tests defined in [test_clusterstate](https://github.com/mesosphere/dcos-enterprise/blob/master/packages/dcos-integration-test/extra/test_clusterstate.py) under dcos-enterprise repo also need our care.

Comment (mainred) [reply packages/dcos-integration-test/extra/test_composition.py]: We may add an additional commit to fix this based on the ones pushed by mergebot.

Comment (wavesoft) [reply packages/dcos-integration-test/extra/test_composition.py]: Oh I thought this list was appended to the enterprise component. I didn't notice that we have a separate list for the EE components. Will check now 👍

File: packages/dcos-integration-test/extra/test_networking.py (hunk 3 above)
Comment (wavesoft) [reply packages/dcos-integration-test/extra/test_networking.py]: Will add, however I have a feeling we might forget that comment after we add IPV6 support 😄

**Comment by wavesoft** [comment]
@mesosphere-mergebot changelog-not-required part of the calico initiative

**Comment by wavesoft** [comment]
@mesosphere-mergebot sync

**Pull request merged**